### PR TITLE
Ensure Checkout block supports the theme-level global styles for `Colors » Buttons`

### LINKED ISSUE #8808: Ensure Checkout block supports the theme-level global styles for `Colors » Buttons`
Currently, the Checkout block partially supports the default theme-level global styles for `Colors » Buttons`. This issue aims to ensure that the Checkout blocks fully support the default theme-level global styles.

## Steps to reproduce

1. Go to `WP Admin » Apperance » Editor`.
2. Open the `Styles` sidebar.
3. Open the `Colors » Buttons` settings.
4. Adjust the settings `TEXT COLOR` and `BACKGROUND COLOR`.
5. See that some settings are not supported.

## Screenshots

<table>
<tr>
<td>Checkout:
<br><br>
Colors-Buttons-Checkout
</td>
</tr>
</table>

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #8808

Currently, the Checkout block partially supports the default theme-level global styles for `Colors » Buttons`. While the `Place Order` button of the Checkout block shows the expected colors in the frontend, it does not show the expected colors in the editor. This PR aims to ensure that the default theme-level global styles for `Colors » Buttons` work both in the editor and the frontend. 

### Additional note

Currently, the default theme-level global styles for `Colors » Buttons` only allow adjusting the text color and the background color. Adjusting the hover colors is not possible (yet).

### Screenshots

<table>
<tr>
<td>Before:
<br><br>

Screenshot 2023-03-24 at 14 06 09

</td>
<td>After:
<br><br>

Screenshot 2023-03-24 at 14 07 04

</td>
</tr>
</table>

#### User Facing Testing

1. Install and activate a block theme, e.g. [Twenty Twenty-Three](https://wordpress.org/themes/twentytwentythree/).
3. Create a test page with the Checkout block.
4. Go to `WP Admin » Apperance » Editor`.
5. Open the `Styles` sidebar.
6. Open the `Colors » Buttons` settings.
7. Adjust the settings `TEXT COLOR` and `BACKGROUND COLOR`.
8. Open the test page from step 2. both in the editor and the frontend.
9. Verify that the `Place Order` button shows the colors as defined in step 6.

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Enhancement: Ensure Checkout block supports the theme-level global styles for `Colors » Buttons`.

### CODE REVIEW

File: assets/js/base/components/button/style.scss [hunk 1]
```diff
@@ -54,6 +54,12 @@
 			opacity: 1;
 		}
 	}
+
+	&.contained {
+		&:hover {
+			opacity: 0.9;
```
Comment (tarunvijwani) [inline assets/js/base/components/button/style.scss]: We're not changing the opacity for Mini Cart and Add to Cart buttons.
I think we should remove this and keep it consistent with core button blocks. 

Slack: p1680026511680059/1680024656.322559-slack-C02UBB1EPEF

File: assets/js/base/components/button/style.scss [hunk 2]
```diff
@@ -79,17 +85,6 @@ body:not(.woocommerce-block-theme-has-button-styles) .wc-block-components-button
...
-		&:disabled,
-		&:hover,
-		&:focus,
-		&:active {
-			background-color: $gray-900;
-			color: $white;
-		}
-
 		&:hover {
 			opacity: 0.9;
```
Comment (tarunvijwani) [inline assets/js/base/components/button/style.scss]: We should also remove this and keep it consistent with core button blocks. We need to double-check the existing styles to ensure we're not overriding any style like this. 

Slack: p1680026511680059/1680024656.322559-slack-C02UBB1EPEF

**Review by tarunvijwani — CHANGES_REQUESTED** [review]
Thank you for the quick fix! I have added the a few suggestion related to opacity. I think we need to double check it the design team. Could you please share this with TJ and confirm is it okay to change the opacity?

File: assets/js/base/components/button/style.scss (hunk 1 above)
Comment (nielslange) [reply assets/js/base/components/button/style.scss]: I've added this, to ensure that the hover state for the buttons on the Checkout block is consistent both in the post editor and the frontend. Based on the Slack discussion, you mentioned, I removed that hover state.

File: assets/js/base/components/button/style.scss (hunk 2 above)
Comment (nielslange) [reply assets/js/base/components/button/style.scss]: Similar to [the hover state before](https://github.com/woocommerce/woocommerce-blocks/pull/8844#discussion_r1151348363), I also removed this one here.

**Comment by nielslange** [comment]
Thanks for your review, @tarunvijwani. I just applied all the suggestions. Would you mind doing another review?

**Review by tarhi-saad — APPROVED** [review]
Thank you, @nielslange, for working on this PR! The test was successful!

**Comment by nielslange** [comment]
Thanks for reviewing this PR, @tarhi-saad! 🙏

**Pull request merged**